crates/uv/tests/it/pip_compile.rs

```diff
@@ -13270,7 +13270,9 @@ exceptiongroup==1.0.0rc8
     #    uv pip compile --cache-dir [CACHE_DIR] requirements.in -c constraints.txt --universal -p 3.10
     alembic==1.8.1
         # via -r requirements.in
-    astroid==2.13.5
+    astroid==2.13.5 ; python_full_version >= '3.11'
+        # via pylint
+    astroid==3.1.0 ; python_full_version < '3.11'
```

Comment:
This isn't _wrong_ but perhaps a bit surprising.

Reply:
This is normal to see for this resolution, it's an artifact of pinning on so many libraries that are a bit old and then trying to resolve them for newer versions of Python.

The correct solution here is that the user (me) shouldn't pin so much, but this is an older set of requirements I use to smoke test new versions of uv, because I know the requirements are challenging.